src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,206 @@
+        String queryUpdateCar = "UPDATE cars SET model=?, manufacturer_id=? WHERE id=? "
+                + "AND deleted=FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stUpdateCar = connection
+                        .prepareStatement(queryUpdateCar)) {
+            stUpdateCar.setString(1, car.getModel());
+            stUpdateCar.setLong(2, car.getManufacturer().getId());
+            stUpdateCar.setLong(3, car.getId());
+            stUpdateCar.executeUpdate();
+            stUpdateCar.close();
+            deleteOldDrivers(car, connection);
+            insertNewDrivers(car, connection);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String queryDelete = "UPDATE cars SET deleted=TRUE WHERE id=?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stDelete = connection.prepareStatement(queryDelete)) {
+            stDelete.setLong(1, id);
+            return stDelete.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id " + id, e);
+        }
+    }
+
+    private void deleteOldDrivers(Car car, Connection connection) throws SQLException {
+        String queryDeleteOldDrivers = "DELETE FROM cars_drivers WHERE car_id=?";
+        PreparedStatement stDeleteDrivers = connection
+                .prepareStatement(queryDeleteOldDrivers);
+        stDeleteDrivers.setLong(1, car.getId());
+        stDeleteDrivers.executeUpdate();
+        stDeleteDrivers.close();
+    }
+
+    private void insertNewDrivers(Car car, Connection connection) throws SQLException {
+        String queryInsertNewDrivers = "INSERT INTO cars_drivers (driver_id, car_id)"
+                + "VALUES(?, ?)";
+        PreparedStatement stInsertDrivers = connection
+                .prepareStatement(queryInsertNewDrivers);
+        stInsertDrivers.setLong(2, car.getId());
+        for (Driver driver : car.getDrivers()) {
+            stInsertDrivers.setLong(1, driver.getId());
+            stInsertDrivers.executeUpdate();
+        }
+        stInsertDrivers.close();
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        car.setManufacturer(getManufacturerById(resultSet.getObject("manufacturer_id",
+                Long.class)));
+        List<Driver> drivers = new ArrayList<>();
+        while (resultSet.next()) {
+            drivers.add(getDriverById(resultSet.getObject("driver_id", Long.class)));
+        }
+        car.setDrivers(drivers);
+        return car;
+    }
+
+    private Manufacturer getManufacturerById(Long id) {
```

Comment:
in this method and in method getDriverById() you open new connections

this means that there are multiple open connections inside each other (because you use these methods in get(), getAll() and so on)

we should refactor this